packages/usa-input-mask/src/content/usa-input-mask-date-short.json

```diff
@@ -0,0 +1,10 @@
+{
+  "hint": "For example, 01/01/1970",
+  "id": "us-date",
+  "inputmode": "numeric",
+  "label": "US Date",
+  "mask": "99/99/9999",
+  "pattern": "\\d{2}/\\d{2}/\\d{4}",
```

Comment:
I wonder if this accepted pattern could be more constrained so that users can't input an invalid date like 33/33/3333? Not sure if that is out of scope...

Reply:
Hi Amy! Thanks for commenting on this... We could and maybe should do this! I wanted to explain a little on why this would be "separate" from the inner workings of the input mask JS and why, even with the pattern set to force an actual valid date, a user would still be able to type in 33/33/3333.

The pattern attribute only affects CSS (which is currently not implemented or represented in the examples) and is not needed for the input mask to work. (what's really used is the mask attribute 99/99/9999 in this case, which only allows numbers).

In other words, if we added a more complex regex pattern check, it would NOT affect or limit the mask in anyway, it would just trigger the `:invalid` pseudo class as "true" (meaning someone could implement a red border or background on the field as instant feedback). It would still allow for a date like 33/33/3333 but would have a red border.

To truly check for a valid date (like disallowing 13 as a month), I believe that would have to be done in the JS and may be out of scope for this version.

This makes me wonder if we should implement the `:invalid` CSS and recommend its use like the following for all mask examples:

```
input:invalid {
  border: red solid 3px;
}
```

Source: https://developer.mozilla.org/en-US/docs/Web/HTML/Attributes/pattern

I'll bring it up to Dan at our sync today, thanks again!

Reply:
Thanks for the explanation! It all makes sense. It makes me wonder if the input mask is an appropriate solution for Date, or if it might be better to point users to the Date picker or memorable date components for that kind of entry. Would be curious to hear your thoughts!

Reply:
The sense we got from Dan on this is that we shouldn't try to build in more intelligence e.g. to validate a date. A few reasons include:
- keeping it simple (e.g. only validate if it's a letter or number)
- having to potentially account for a variety of other date formats (e.g. if a user wanted to put dd/mm/yyyy or mm/yy etc.)
- would we need to build in intelligence for other masks (e.g. area codes, SSNs etc. and limit it to only valid inputs)

As such, we are actually going to remove the `pattern` attribute all together, as well as, the new pattern guidance as it may add some confusion to implementing a mask. Updates coming soon!